src/components/SheetBuilder/SheetBuilderForm/SheetBuilderFormSelect.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import Select from 'react-select';
+import { selectStyles } from '../common/selectStyles';
+
+type Value<T, IsMulti extends boolean> = IsMulti extends true
+  ? readonly T[]
+  : T;
+
+type Props<T, IsMulti extends boolean, V = Value<T, IsMulti>> = Record<
+  string,
+  any
+> & {
+  options: T[];
+  isMulti?: IsMulti;
+  onChange: (newValue: V) => void;
+};
+
+const SheetBuilderFormSelect = <T, IsMulti extends boolean = false>({
+  options,
+  onChange,
+  placeholder,
+  className,
+  isSearcheable,
+  isMulti,
+  id,
+  isClearable,
+  defaultValue,
+}: Props<T, IsMulti>) => (
+  <Select
+    options={options}
+    styles={selectStyles}
+    onChange={(newValue) => onChange(newValue)}
+    placeholder={placeholder}
+    className={className}
+    isSearchable={isSearcheable ?? false}
+    isMulti={isMulti}
+    instanceId={id}
+    isClearable={isClearable}
+    defaultValue={defaultValue}
+  />
+);
```

Comment:
Uma vez tendo os tipos definidos corretamente, você não precisa explicitamente passar esses componentes como props para o componente Select, já que vão estar bem descritos na tipagem, você pode fazer um {...props} e passar explicitamente só oq não vem de prop. 

```suggestion
const SheetBuilderFormSelect = <T, IsMulti extends boolean = false>({
  options,
  onChange,
  ...props
}: Props<T, IsMulti>) => (
  <Select
    options={options}
    styles={selectStyles}
    onChange={newValue => onChange(newValue)}
    {...props}
  />
);
```